R/MsExperiment.R

```diff
@@ -21,60 +21,67 @@ setMethod("saveMsObject",
               ## sample data
               write.table(as.data.frame(object@sampleData), sep = "\t",
                           file = file.path(param@path,
-                                           "sample_data.txt"))
+                                           "ms_experiment_sample_data.txt"))
 
-              ## sample data links ## maybe here is better if we create a separate folder for all these.
+              ## sample data links
               sdl <- object@sampleDataLinks
-              lapply(names(sdl), function(x){
-                  fl <- file.path(param@path, paste0("sample_data_links_", x,
-                                                     ".txt"))
-                  write.table(sdl[[x]], file = fl, row.names = FALSE,
-                              col.names = FALSE, sep = "\t")
+              if (length(sdl) > 0) {
+                  lapply(names(sdl), function(x){
+                      fl <- file.path(
+                          param@path,
+                          paste0("ms_experiment_sample_data_links_", x, ".txt"))
+                      write.table(sdl[[x]], file = fl, row.names = FALSE,
+                                  col.names = FALSE, sep = "\t")
                   })
-              write.table(sdl@elementMetadata, sep = "\t",
-                          file = file.path(param@path, "element_metadata.txt"))
+                  write.table(
+                      sdl@elementMetadata, sep = "\t", quote = TRUE,
+                      file = file.path(param@path,
+                                       "ms_experiment_link_mcols.txt"))
+              }
               ## call export of individual other objects (not MsExperiment data)
               saveMsObject(spectra(object), param)
               ## at some point also chromatograms, etc.
           }
 )
 
 #' @rdname PlainTextParam
+#'
+#' @importMethodsFrom S4Vectors mcols<-
 setMethod("readMsObject",
           signature(object = "MsExperiment",
                     param = "PlainTextParam"),
-          function(object, param, spectraPath = character()) {
+          function(object, param, ...) {
               ## read sample data
-              fl <- file.path(param@path, "sample_data.txt")
+              fl <- file.path(param@path, "ms_experiment_sample_data.txt")
               if (!file.exists(fl))
-                  stop("No 'sample_data.txt' file found in the provided path.")
+                  stop("No 'ms_experiment_sample_data.txt' file found in ",
+                       "the provided path.")
               sd <- read.table(fl, sep = "\t")
               object@sampleData <- DataFrame(sd, row.names = NULL)
 
               ## read spectra
-              object@spectra <- readMsObject(Spectra::Spectra(), param, spectraPath = spectraPath)
-
+              if (file.exists(file.path(param@path, "spectra_slots.txt")))
```

Comment:
Soo we support import even if we don't have spectra files ? we went a bit back and forth on this i think

Reply:
In the long run (now apart from the use case we have in *xcms*) it would be good to support also saving an `MsExperiment` without `Spectra`. I would expect that that should also work with this improved implementations you did (compared to the initial prototype we had in *xcms*).